### PR TITLE
Update ref to 3 scale dev deployment

### PR DESCRIPTION
**Pull request by casey-williams-rh**
#### Jira Ticket: https://issues.redhat.com/browse/RHCLOUD-20960
----

## What?
Updating references from the old 3scale Dev and Stage deployments `apicast.3scale-(dev|stage).svc.cluster.local` to the new 3scale dev deployment `gateway.3scale-(dev|stage).svc.cluster.local`

## Why?
The reason for this change is the Infrastructure Team has stood up a new 3scale (Gateway) Deployment to support service serving certificates. Currently, both the old and new deployments are running in parallel so we can aid workstreams in the change over.

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: deploy/template.yml [hunk 1]
```diff
@@ -569,7 +569,7 @@ parameters:
   value: bop
 - name: AUTH_GATEWAY_URL
   required: true
-  value: "http://apicast.3scale-staging.svc.cluster.local:8890/internal/certauth"
```
Comment (casey-williams-rh) [inline deploy/template.yml:572]: I noticed that this URL references `staging` not `stage`. Need to confirm this should be pointing to the Stage Cluster (`crcs02ue1`).

File: deploy/template-prod.yml [hunk 2]
```diff
@@ -440,7 +440,7 @@ parameters:
   value: bop
 - name: AUTH_GATEWAY_URL
   required: true
-  value: "http://apicast.3scale-staging.svc.cluster.local:8890/internal/certauth"
```
[the remark this answers is not in the document]
Comment (dehort) [reply deploy/template-prod.yml:443]: Cloud-Connector should be pointed at 3scale running in stage.  Yeah, I think this should be the stage cluster.

File: deploy/template.yml (hunk 1 above)
Comment (dehort) [reply deploy/template.yml:572]: Cloud-Connector should be pointed at 3scale running in stage.  Yeah, I think this should be the stage cluster.

File: internal/config/config.go [hunk 3]
```diff
@@ -284,7 +284,7 @@ func GetConfig() *Config {
 	options.SetDefault(JWT_TOKEN_EXPIRY, 1)
 	options.SetDefault(JWT_PRIVATE_KEY_FILE, "/etc/jwt/mqtt-private-key.rsa")
 	options.SetDefault(JWT_PUBLIC_KEY_FILE, "/etc/jwt/mqtt-public-key.rsa")
-	options.SetDefault(AUTH_GATEWAY_URL, "http://apicast.3scale-staging.svc.cluster.local:8890/internal/certauth")
```
[the remark this answers is not in the document]
Comment (dehort) [reply internal/config/config.go:287]: Cloud-Connector should be pointed at 3scale running in stage.  Yeah, I think this should be the stage cluster.

**Review by dehort — APPROVED** [review]

**Pull request merged**